servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/AbstractRestContents.java

```diff
@@ -163,24 +210,129 @@ public void verifyContentAndOperationTypesIndividually(
+                .element(0)
+                // Clear content ID for comparison
+                .extracting(this::clearIdOnOperation)
+                .isEqualTo(put);
+          });
+    } else {
+      // not a Put operation
+      assertAll(
+          () -> {
+            List<Entry> entries =
+                getApi().getEntries().refName(branch.getName()).get().getEntries();
+            assertThat(entries).isEmpty();
+          },
+          () -> {
+            // Diff against of committed HEAD and previous commit must yield the content in the
+            // Put operations
+            assertThat(getApi().getDiff().fromRef(committed).toRef(branch).get())
+                .extracting(DiffResponse::getDiffs, list(DiffEntry.class))
+                .isEmpty();
+          },
+          () -> {
+            // Compare content on HEAD commit with the committed content
+            Map<ContentKey, Content> content =
+                getApi().getContent().key(fixedContentKey).reference(committed).get();
+            assertThat(content).isEmpty();
+          },
+          () -> {
+            // Compare operation on HEAD commit with the committed operation
+            LogResponse log =
+                getApi().getCommitLog().reference(committed).fetch(FetchOption.ALL).get();
+            assertThat(log)
+                .extracting(LogResponse::getLogEntries, list(LogEntry.class))
+                .element(0)
+                .extracting(LogEntry::getOperations)
+                .satisfies(
+                    ops -> {
+                      if (contentAndOperationType.operation instanceof Delete) {
+                        // Delete ops are persisted - must occur in commit
+                        assertThat(ops).containsExactly(contentAndOperationType.operation);
+                      } else if (contentAndOperationType.operation instanceof Unchanged) {
+                        // Unchanged ops are not persisted - cannot occur in commit
+                        assertThat(ops).isNullOrEmpty();
+                      } else {
+                        fail("Unexpected operation " + contentAndOperationType.operation);
+                      }
+                    });
+          });
+    }
+  }
+
+  private List<Operation> clearIdOnOperations(List<Operation> o) {
+    return o.stream().map(this::clearIdOnOperation).collect(Collectors.toList());
+  }
+
+  private Operation clearIdOnOperation(Operation o) {
+    try {
+      if (!(o instanceof Put)) {
+        return o;
+      }
+      Put put = (Put) o;
+      Content contentWithoutId = clearIdOnContent(put.getContent());
+      return ImmutablePut.builder().from(put).content(contentWithoutId).build();
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private Content clearIdOnContent(Content content) {
+    try {
+      return (Content)
+          content
+              .getClass()
+              .getDeclaredMethod("withId", String.class)
```

Comment:
maybe move `withId` to a utility class so that we don't have to use reflection. Also should this rather be `withoutId`?

Reply:
`withId` is generated code in all Immutable* classes that extend `Content`. Don't like to have a publicly accessible functionality to update or clear the content ID.

Reply:
imo it would be good enough to move such functionality into a test utility class